aloscene/scene_flow.py

```diff
@@ -0,0 +1,183 @@
+import aloscene
+from aloscene import Depth, CameraIntrinsic, Flow, Mask
+from typing import Union
+import numpy as np
+import torch
+import torch.nn.functional as F
+
+
+def load_scene_flow(path: str) -> torch.Tensor:
+    with open(path, "rb") as file:
+        data = np.load(file)
+        return torch.from_numpy(data)
+
+
+class SceneFlow(aloscene.tensors.SpatialAugmentedTensor):
+    """
+    Scene flow map
+
+    Parameters
+    ----------
+    x : str
+        load scene flow from a numpy file
+    """
+
+    @staticmethod
+    def __new__(cls, x, occlusion: Mask = None, *args, names=("C", "H", "W"), **kwargs):
+        if isinstance(x, str):
+            # load flow from path
+            x = load_scene_flow(x)
+            names = ("C", "H", "W")
+
+        tensor = super().__new__(cls, x, *args, names=names, **kwargs)
+        tensor.add_child("occlusion", occlusion, align_dim=["B", "T"], mergeable=True)
+        return tensor
+
+    def __init__(self, x, *args, **kwargs):
+        super().__init__(x)
+
+    @classmethod
+    def from_optical_flow(
+        cls,
+        optical_flow: Flow,
+        depth: Depth,
+        next_depth: Depth,
+        intrinsic: CameraIntrinsic,
+    ):
+        """Create scene flow from optical flow, depth a T, depth at T + 1 and the intrinsic
+
+        Parameters
+        ----------
+        optical flow: aloscene.Flow
+            The optical flow at T.
+        depth: aloscene.Depth
+            The depth at T.
+        next_depth: aloscene.Depth
+            The depth at T + 1
+        intrinsic : aloscene.CameraIntrinsic
+            The intrinsic of the image at T.
+        """
+
+        has_batch = True if len(optical_flow.names) == 4 else False
+
+        if optical_flow.names != depth.names or optical_flow.names != next_depth.names:
+            raise ValueError(
+                "The optical flow, depth and next_depth must have the same names"
+            )
+
+        if optical_flow.names != ("C", "H", "W") and optical_flow.names != (
+            "B",
+            "C",
+            "H",
+            "W",
+        ):
```

Comment:
remove the last comma in `("B", "C", "H", "W",)` to avoid newlines in auto-formatting